File: ImperatorToCK3Tests/MapperTests/ProvinceMapper/ProvinceMappingsVersionTests.cpp
#include "../../../ImperatorToCK3/Source/Mappers/ProvinceMapper/ProvinceMappingsVersion.h"
#include "gtest/gtest.h"
#include <sstream>
#include <string>


TEST(Mappers_ProvinceMappingsVersionTests, MappingsDefaultToEmpty)
{
	std::stringstream input;
	input << "= {\n";
	input << "}";

	const mappers::ProvinceMappingsVersion theMappingVersion(input);

	ASSERT_TRUE(theMappingVersion.getMappings().empty());
}

TEST(Mappers_ProvinceMappingsVersionTests, MappingsCanBeLoaded)
{
	std::stringstream input;
	input << "= {\n";
	input << "	link = { ck3 = 1 imp = 1 }\n";
	input << "	link = { ck3 = 2 imp = 2 }\n";
	input << "}";

	const mappers::ProvinceMappingsVersion theMappingVersion(input);

	ASSERT_EQ(2, theMappingVersion.getMappings().size());
}

File: ImperatorToCK3.UnitTests/Mappers/Province/ProvinceMappingsVersionTests.cs
using commonItems;
using ImperatorToCK3.Mappers.Province;
using Xunit;

namespace ImperatorToCK3.UnitTests.Mappers.Province; 

public class ProvinceMappingsVersionTests {
	[Fact]
	public void MappingsDefaultToEmpty() {
		var reader = new BufferedReader(
			"= {}"
		);
		var theMappingVersion = new ProvinceMappingsVersion(reader);
		Assert.Empty(theMappingVersion.Mappings);
	}

	[Fact]
	public void MappingsCanBeLoaded() {
		var reader = new BufferedReader(
			"= {\n" +
			"	link = { ck3 = 1 imp = 1 }\n" +
			"	link = { ck3 = 2 imp = 2 }\n" +
			"}"
		);
		var theMappingVersion = new ProvinceMappingsVersion(reader);
		Assert.Equal(2, theMappingVersion.Mappings.Count);
	}
}